<?xml version="1.0"?>
<launch>

  	<arg name="robotName" default="ropod"/>
	<arg name="laser1_name" value ="laser"/>
<!--    <arg name="map_file" default="$(find ed_object_models)/models/hospital_test_amcl/walls/map_amcl.yaml" />-->
<!--	<arg name="map_file" default="$(find ropod_navigation_test)/config/amk-f064/map.yaml"/>-->
<!--    <arg name="map_file" default="/home/cesar/ropod-project-software/catkin_workspace/src/applications/occupancy_grids/tue/lab-level/map.yaml"/>-->
<!--    <arg name="map_file" default="$(find ed_object_models)/models/tue_hallway_RLtoAL/walls/shape/map.yaml" />-->
    <arg name="map_file" default="$(find ed_object_models)/models/tue_hallway_RLtoAL/walls/shape/map_2DoorsOpen.yaml" />
<!--    <arg name="map_file" default="$(find ed_object_models)/models/brsu/walls/shape/map.yaml" /> -->
<!--        <arg name="map_file" default="/home/bob/ropod-project-software/catkin_workspace/src/applications/occupancy_grids/brsu/brsu-c-floor0_osm/map.yaml"/>-->

   <!-- Run the map server -->
     <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)" />

    <!--- Run AMCL -->
    <node pkg="amcl" type="amcl" name="amcl">
        <remap from="scan" to="/$(arg robotName)/$(arg laser1_name)/scan"/>
        
        <!-- <remap from="scan" to="/projected_scan_front"/> --> <!-- for BRSU ropod -->

        <!--- Odometery model parameters  -->
        <param name="odom_model_type" value="omni"/>
        <param name="odom_alpha1" value="7.0"/>
        <param name="odom_alpha2" value="5.0"/>
        <param name="odom_alpha3" value="5.0"/>
        <param name="odom_alpha4" value="5.0"/>
        <param name="odom_alpha5" value="7.0"/>
        <param name="odom_frame_id" value="$(arg robotName)/odom"/>
        <param name="base_frame_id" value="$(arg robotName)/base_link"/>
        <param name="global_frame_id" value="/map"/>

        <!-- Overall filter parameters -->
        <param name="min_particles" value="200"/>
        <param name="max_particles" value="2000"/>
        <param name="kld_err" value="0.01"/>
        <param name="kld_z" value="0.99"/>
        <param name="update_min_d" value="0.05"/>
        <param name="update_min_a" value="0.01"/>
        <param name="resample_interval" value="2"/>
        <param name="transform_tolerance" value="0.1" />
        <param name="recovery_alpha_slow" value="0.0"/>
        <param name="recovery_alpha_fast" value="0.0"/>
        <param name="gui_publish_rate" value="-1"/>
        <param name="use_map_topic" value="true"/>
        <param name="first_map_only" value="true"/>

        <!--Laser model parameters -->
        <param name="laser_min_range" value="0.04"/>
        <param name="laser_max_range" value="5.0"/>
        <param name="laser_max_beams" value="100"/>
        <param name="laser_z_hit" value="0.95"/>
        <param name="laser_z_short" value="0.01"/>
        <param name="laser_z_max" value="0.05"/>
        <param name="laser_z_rand" value="0.05"/>
        <param name="laser_sigma_hit" value="0.2"/>
        <param name="laser_lambda_short" value="0.1"/>
        <param name="laser_likelihood_max_dist" value="1.0"/>
        <param name="laser_model_type" value="likelihood_field"/>

<!--        initial covariance values-->
        <param name="initial_cov_xx" value="1"/>
        <param name="initial_cov_yy" value="1"/>
        <param name="initial_cov_aa" value="0.5"/>

        <!-- initial pose of the robot in TU/e hallway-->
        <param name="initial_pose_x" value="-1.6"/>
        <param name="initial_pose_y" value="2.2"/>
        <param name="initial_pose_a" value="0"/>

        <!-- initial pose of the robot in BRSU-->
        <!--<param name="initial_pose_x" value="59.74"/>-->
        <!--<param name="initial_pose_y" value="32.36"/>-->
        <!--<param name="initial_pose_a" value="-3.14"/>-->

    </node>

    % Launch ED
<!--    <node pkg="ed" type="ed" name="ed" args="$(find ropod_navigation_test)/config/model-example-ropod-napoleon-navigation-ED.yaml"> </node>-->

    <node name="napoleon_navigation" pkg="napoleon_navigation" type="napoleon_navigation" output="screen">

<!--        <rosparam file="$(find ropod_navigation_test)/config/parameters/costmap_common_params.yaml" command="load" ns="local_costmap" />-->
<!--        <rosparam file="$(find ropod_navigation_test)/config/parameters/footprint_ropod.yaml" command="load" ns="local_costmap" />-->
<!--        <rosparam file="$(find ropod_navigation_test)/config/parameters/local_costmap_params.yaml"  command="load"/>-->
<!--        <rosparam file="$(find ropod_navigation_test)/config/parameters/teb_local_planner_params_ropod.yaml" command="load" />-->

        <rosparam param="footprint_param" >[ 0,0.175, 0.45,0.175, 0.45,0, 2.69,0, 2.69,0.175, 2.81,0.175, 2.81,0.825, 2.69,0.825, 2.69,1, 0.45,1, 0.45,0.825, 0,0.825]</rosparam> 
        <rosparam param="footprintMiddlePose_param" >[ 1.405,0.5,3.1415 ]</rosparam>
        <rosparam param="wheelDistanceMiddle_param" > 1 </rosparam>
<!--        <rosparam param="footprint_param" >[ 0,0, 0.65,0, 0.65,0.65, 0,0.65 ]</rosparam>-->
<!--        <rosparam param="footprintMiddlePose_param" >[ 0.325,0.325,0 ]</rosparam>-->
<!--        <rosparam param="wheelDistanceMiddle_param" > 0.3 </rosparam>-->

        <rosparam param="maxSpeed_param" > 1.4 </rosparam>
        <rosparam param="maxAcceleration_param" > 0.7 </rosparam>

        <rosparam param="tubeWallOffset_param" > 0.1 </rosparam>
        <rosparam param="tubeExtraSpace_param" > 0.2 </rosparam>

        <rosparam param="nTries_param" > 10 </rosparam>
        <rosparam param="predictionTime_param" > 4 </rosparam>
        <rosparam param="minPredictionDistance_param" > 0 </rosparam>
        <rosparam param="predictionBiasFactor_param" > 0.3 </rosparam>

        <remap from="/navigation/cmd_vel" to="/$(arg robotName)/cmd_vel"/>
        <remap from="/navigation/odom" to="/$(arg robotName)/odom"/>
        <remap from="/navigation/scan" to="/$(arg robotName)/$(arg laser1_name)/scan"/>

        <param name="odom_frame_id" value="$(arg robotName)/odom"/>
<!--        <param name="base_frame_id" value="$(arg robotName)/base_link"/>-->
        <param name="base_frame_id" value="$(arg robotName)/flock_base"/>
        <param name="global_frame_id" value="/map"/>

    </node>


</launch>
